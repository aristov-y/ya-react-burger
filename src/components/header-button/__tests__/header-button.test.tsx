import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import HeaderButton from '../header-button';

jest.mock('@ya.praktikum/react-developer-burger-ui-components', () => ({
  Button: ({ children }) => (<div data-id="Button">{children}</div>)
}));

jest.mock('../getIcon', () => (type) => `Icon:${type}`);

describe('HeaderButton', function () {
  it('should render', function () {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<HeaderButton text={'ButtonText'} type={'BURGER'} to={'to-burger'} />)
    });

    expect(component.toJSON()).toMatchSnapshot();

    act(() => {
      component.update(<HeaderButton text={'Text'} type={'PROFILE'} to={'to-profile'} />)
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
