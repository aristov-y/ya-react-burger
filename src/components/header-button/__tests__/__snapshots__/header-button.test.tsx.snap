// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton should render 1`] = `
<div
  data-id="NavLink"
>
  <div
    data-id="Button"
  >
    Icon:BURGER
    <span
      className="text ml-2 text_type_main-default ButtonText"
      style={
        Object {
          "verticalAlign": "super",
        }
      }
    >
      ButtonText
    </span>
  </div>
</div>
`;

exports[`HeaderButton should render 2`] = `
<div
  data-id="NavLink"
>
  <div
    data-id="Button"
  >
    Icon:PROFILE
    <span
      className="text ml-2 text_type_main-default ButtonText"
      style={
        Object {
          "verticalAlign": "super",
        }
      }
    >
      Text
    </span>
  </div>
</div>
`;
